### PR TITLE
[frameworks] validate getOutputDirName and other missing values on framework entries

### PR DESCRIPTION
**Pull request by EndangeredMassa**
Covers the validation in Framework entries over function properties and other missing holes.

### CODE REVIEW

File: packages/frameworks/test/frameworks.unit.test.ts [hunk 1]
```diff
@@ -62,16 +62,40 @@ const SchemaSettings = {
...
+    },
+  },
+};
+
 const Schema = {
   type: 'array',
   items: {
     type: 'object',
-    required: ['name', 'slug', 'logo', 'description', 'settings'],
+    additionalProperties: false,
```
Comment (EndangeredMassa) [inline packages/frameworks/test/frameworks.unit.test.ts:83]: Adding a new property to a framework entry will now cause this test to fail, forcing the author to add a type to the schema for it.

File: packages/frameworks/test/frameworks.unit.test.ts [hunk 2]
```diff
@@ -62,16 +62,40 @@ const SchemaSettings = {
...
+    required: [
+      'name',
+      'slug',
+      'logo',
+      'description',
+      'settings',
+      'getOutputDirName',
```
Comment (EndangeredMassa) [inline packages/frameworks/test/frameworks.unit.test.ts:90]: `getOutputDirName` is now required because it's definitely used during deploy

**Review by styfle — DISMISSED** [review]
Very nice, thanks! 🎉

File: packages/frameworks/test/frameworks.unit.test.ts [hunk 3]
```diff
@@ -246,3 +291,16 @@ describe('frameworks', () => {
     );
   });
 });
+
+function getValidator() {
+  const ajv = new Ajv();
+
+  ajv.addKeyword('isFunction', {
+    compile: shouldMatch => data => {
+      const matches = data instanceof Function;
```
Comment (TooTallNate) [inline packages/frameworks/test/frameworks.unit.test.ts]: I'd prefer `typeof data === 'function'`, but not a huge deal.

**Review by TooTallNate — DISMISSED** [review]

File: packages/frameworks/test/frameworks.unit.test.ts (hunk 3 above)
Comment (EndangeredMassa) [reply packages/frameworks/test/frameworks.unit.test.ts]: I like that better as well. [`bed27ff` (#9900)](https://github.com/vercel/vercel/pull/9900/commits/bed27ffd21c2cd7de6783452f9491b7241f34dc0)

**Review by TooTallNate — APPROVED** [review]

**Review by cb1kenobi — APPROVED** [review]

**Review by styfle — APPROVED** [review]

**Pull request merged**